samples/client/petstore/go/go-petstore/pet_api.go

```diff
@@ -1,602 +1,566 @@
-  localVarHttpContentType := a.Configuration.APIClient.SelectHeaderContentType(localVarHttpContentTypes)
-  if localVarHttpContentType != "" {    
-      headerParams["Content-Type"] = localVarHttpContentType
-  }
-  // to determine the Accept header
-  localVarHttpHeaderAccepts := []string {
-      "application/json", 
-  }
-  //set Accept header
-  localVarHttpHeaderAccept := a.Configuration.APIClient.SelectHeaderAccept(localVarHttpHeaderAccepts)
-  if localVarHttpHeaderAccept != "" {  
-      headerParams["Accept"] = localVarHttpHeaderAccept
-  }
-
-  formParams["additionalMetadata"] = additionalMetadata
-  fbs, _ := ioutil.ReadAll(file)
-  fileBytes = fbs
-  fileName = file.Name()
-
-  var successPayload = new(ModelApiResponse)
-  httpResponse, err := a.Configuration.APIClient.CallAPI(path, httpMethod, postBody, headerParams, queryParams, formParams, fileName, fileBytes)
-
-
-  if err != nil {
-    return *successPayload, *NewAPIResponse(httpResponse.RawResponse), err
-  }
-
-  err = json.Unmarshal(httpResponse.Body(), &successPayload)
-
-  return *successPayload, *NewAPIResponse(httpResponse.RawResponse), err
+func (a PetApi) UploadFile(petId int64, additionalMetadata string, file *os.File) (ModelApiResponse, APIResponse, error) {
+
+	var httpMethod = "Post"
+	// create path and map variables
+	path := a.Configuration.BasePath + "/pet/{petId}/uploadImage"
+	path = strings.Replace(path, "{"+"petId"+"}", fmt.Sprintf("%v", petId), -1)
+
+	// verify the required parameter 'petId' is set
+	if &petId == nil {
+		return *new(ModelApiResponse), *NewAPIResponseWithError("400 - Bad Request"), errors.New("Missing required parameter 'petId' when calling PetApi->UploadFile")
+	}
+
+	headerParams := make(map[string]string)
+	queryParams := make(map[string]string)
+	formParams := make(map[string]string)
+	var postBody interface{}
+	var fileName string
+	var fileBytes []byte
+	// authentication (petstore_auth) required
+
+	// oauth required
+	if a.Configuration.AccessToken != ""{
+		headerParams["Authorization"] =  "Bearer " + a.Configuration.AccessToken
+	}
+	// add default headers if any
+	for key := range a.Configuration.DefaultHeader {
+		headerParams[key] = a.Configuration.DefaultHeader[key]
+	}
+
+	// to determine the Content-Type header
+	localVarHttpContentTypes := []string{ "multipart/form-data",  }
+
+	// set Content-Type header
+	localVarHttpContentType := a.Configuration.APIClient.SelectHeaderContentType(localVarHttpContentTypes)
+	if localVarHttpContentType != "" {
+		headerParams["Content-Type"] = localVarHttpContentType
+	}
+	// to determine the Accept header
+	localVarHttpHeaderAccepts := []string{
+		"application/json",
+	}
+
+	// set Accept header
+	localVarHttpHeaderAccept := a.Configuration.APIClient.SelectHeaderAccept(localVarHttpHeaderAccepts)
+	if localVarHttpHeaderAccept != "" {
+		headerParams["Accept"] = localVarHttpHeaderAccept
+	}
+
+	formParams["additionalMetadata"] = additionalMetadata	fbs, _ := ioutil.ReadAll(file)
```

Comment:
There's a minor issue here. `fbs, _ := ioutil.ReadAll(file)` should be in another line.